caikit_nlp/modules/text_generation/text_generation_task.py

```diff
@@ -0,0 +1,27 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# First party
+from caikit.core import TaskBase, task
+
+# Local
+from ...data_model import GeneratedResult
+
+
+@task(
+    required_parameters={"text": str},
+    output_type=GeneratedResult,
+)
+class TextGenerationTask(TaskBase):
```

Comment:
I need to catch up with caikit task changes, but what information (if anything, assuming @task is basically the secret sauce with parameter signatures etc) is meant to be implemented on this class?

I.e., is this supposed to replace the abstract module type base class?

Reply:
Honestly I am also yet to dig deeper into task logic to understand tasks more. I have the same question.

> is this supposed to replace the abstract module type base class?

What to you mean by this?

Reply:
Is this meant to replace what would be `TextGenerationBase`, or is it completely separate?

Since we never got around to adding it and still inherit from `ModuleBase` directly

Reply:
This is completely separate. It needs to be there for every "type" of module and essentially defines the shape of input / output for a given "task"

Reply:
Tasks allow for the verification of correct run signatures. There can be different modules with the same task.

Reply:
Cool sounds good, just wanted to better understand if the content of the task is largely the decorator, or the decorator + things to be added to the class it's wrapping. Thanks!